frontend/app/model/MembershipAcquisitionData.scala

```diff
@@ -0,0 +1,80 @@
+package model
+
+import cats.syntax.either._
+import com.gu.acquisition.model.{OphanIds, ReferrerAcquisitionData}
+import com.gu.acquisition.typeclasses.AcquisitionSubmissionBuilder
+import com.gu.memsub.{Product => _, _}
+import com.gu.memsub.BillingPeriod.OneOffPeriod
+import com.gu.salesforce.Tier
+import ophan.thrift.event._
+
+case class MembershipAcquisitionData(
+  amountPaidToday: Price,
+  billingPeriod: BillingPeriod,
+  paymentMethod: Option[PaymentMethod],
+  tier: Tier,
+  countryCode: Option[String],
+  pageviewId: Option[String],
+  visitId: Option[String],
+  browserId: Option[String],
+  referrerAcquisitionData: Option[ReferrerAcquisitionData]) {
+
+  import MembershipAcquisitionData._
+
+  val product: Either[String, Product] = Either.fromOption(
+    tierToProduct.get(tier),
+    s"Won't create Acquisition event for ${tier.name} tier"
+  )
+
+  val paymentFrequency: PaymentFrequency = billingPeriodToPaymentFrequency(billingPeriod)
+
+  val paymentProvider: Option[PaymentProvider] = paymentMethod.map(paymentMethodToProvider)
+}
+
+object MembershipAcquisitionData {
+
+  private def billingPeriodToPaymentFrequency(period: BillingPeriod): PaymentFrequency = period match {
+    case BillingPeriod.Month => PaymentFrequency.Monthly
+    case BillingPeriod.Quarter => PaymentFrequency.Quarterly
+    case BillingPeriod.SixMonthsRecurring => PaymentFrequency.SixMonthly
+    case BillingPeriod.Year => PaymentFrequency.Annually
+    case _: OneOffPeriod => PaymentFrequency.OneOff
+  }
+
+  private def paymentMethodToProvider(method: PaymentMethod): PaymentProvider = method match {
+    case _: PaymentCard => PaymentProvider.Stripe
+    case _: PayPalMethod => PaymentProvider.Paypal
+    case _: GoCardless => PaymentProvider.Gocardless
```

Comment:
Do we need to do exhaustive pattern matches here?

Reply:
It is exhaustive - these are the only possible types: https://github.com/guardian/membership-common/blob/master/src/main/scala/com/gu/memsub/PaymentMethod.scala